### PR TITLE
fix(snapzone): fixed highlight to also show submeshes

### PR DESCRIPTION
**Pull request by SimonTheSourcerer**
### Description
Fixed preview for submehses.

- [x] Bug fix (non-breaking change which fixes an issue)

### CODE REVIEW

**Review by Maxsiom — APPROVED** [review]
image

This PR also works perfectly for me. :)

**Review by Maxsiom — CHANGES_REQUESTED** [review]
Okay, too fast. If you save the scene, then change to another scene, and then go back to the saved scene, the highlights get invisible. :(

File: Runtime/Interaction/SnapZone.cs [hunk 1]
```diff
@@ -76,7 +76,6 @@ public Material HighlightMeshMaterial
 
                 return highlightMeshMaterial;
             }
-            set { highlightMeshMaterial = value; }
```
Comment (Gusinuhe) [inline Runtime/Interaction/SnapZone.cs:79]: Why was this removed?

Comment (SimonTheSourcerer) [reply Runtime/Interaction/SnapZone.cs:79]: dunno, you removed it in your PR and it was not used :x

**Comment by SimonTheSourcerer** [comment]
Max's problem is not reproduceable for me, anyone else can test?

**Comment by tomwim** [comment]
> Okay, too fast. If you save the scene, then change to another scene, and then go back to the saved scene, the highlights get invisible. :(

Was able to reproduce and can confirm :/

**Comment by tomwim** [comment]
Downloaded a range of random objects from Turbosquid and created a SnapZone from each of them and then also combined them randomly. Looks good to me.
image
image

File: Runtime/Interaction/SnapZone.cs (hunk 1 above)
Comment (Gusinuhe) [reply Runtime/Interaction/SnapZone.cs:79]: 🤔🤷‍♂️

Comment (Gusinuhe) [reply Runtime/Interaction/SnapZone.cs:79]: No idea, but if this was already published, this is an API break.

**Comment by Gusinuhe** [comment]
I also could not reproduce it

ezgif com-video-to-gif

**Comment by SimonTheSourcerer** [comment]
It depends on the mesh, very complex meshes seems to be loaded longer or something like that. Problem is than, that on Start those meshes have to be loaded. Fixed it by not reloading the meshes every time only when the meshrenderer does not have a mesh.

File: Runtime/Interaction/SnapZone.cs [hunk 2]
```diff
@@ -133,7 +132,10 @@ public Material InvalidMaterial
         
         private Mesh previewMesh;
         
-        internal Mesh PreviewMesh 
+        /// <summary>
+        /// Returns the preview mesh used for this SnapZone.
+        /// </summary>
+        public Mesh PreviewMesh
```
Comment (Gusinuhe) [inline Runtime/Interaction/SnapZone.cs:138]: Does it have to public?

**Review by tomwim — APPROVED** [review]

**Review by Maxsiom — APPROVED** [review]
Works for me now. :)
